4489 Enhance hyperlinks of config syntax doc (#4490)
Project-MONAI/MONAI: docs/source/config_syntax.md

- [A basic example](#a-basic-example)
- [Syntax examples explained](#syntax-examples-explained)

>   - [`@` to interpolate with Python objects](#1--to-interpolate-with-python-objects)

Reviewer: We had discussed the wording here earlier, on further thought perhaps this should be "@ to reference Python objects in configurations" and then using the word reference/referencing elsewhere to describe what @ is doing.

Reviewer 2: sure, sounds good to me.

Author: Hi @ericspod @wyli ,

Updated in the latest commit.
Could you please help review it again?

Thanks in advance.

where `COMMANDS` is one of the following: `run`, `verify_metadata`, `ckpt_export`, ...
(please see `python -m monai.bundle --help` for a list of available options).

> The commands support many fexible use cases, like: override config at runtime, predefine CLI args in a file, etc.

Reviewer 2: ```suggestion
The CLI supports flexible use cases, such as overriding configs at runtime and predefining arguments in a file.
```

Author: updated in the latest commit.

Thanks.

  simple structures with sparse uses of expressions or references are preferred.
- For `$import <module>` in the configuration, please make sure there are instructions for the users to install
  the `<module>` if it is not a (optional) dependency of MONAI.

> - As "#" and "$" can be syntax for `shell` or `CLI` tools, may need to add escape character or quotes for them

Reviewer 2: ```suggestion
- As "#" and "$" might be interpreted differently by the `shell` or `CLI` tools, may need to add escape characters or quotes for them
```
